lightning/src/ln/msgs.rs

```diff
@@ -824,6 +824,27 @@ pub trait RoutingMessageHandler : Send + Sync {
 	fn get_next_node_announcements(&self, starting_point: Option<&PublicKey>, batch_amount: u8) -> Vec<NodeAnnouncement>;
 	/// Returns whether a full sync should be requested from a peer.
 	fn should_request_full_sync(&self, node_id: &PublicKey) -> bool;
+	/// Queries a peer for a list of channels with a funding UTXO in the requested
+	/// chain and range of blocks.
+	fn query_channel_range(&self, their_node_id: &PublicKey, chain_hash: BlockHash, first_blocknum: u32, number_of_blocks: u32) -> Result<(), LightningError>;
```

Comment:
I think these fn interfaces can be cut back a bit - at least chain_hash can be dropped because `NetworkGraph` objects only actually support one chain (though maybe we should track it in the `NetworkGraph`!) and I'm not sure exactly how this would be used so not sure why we need `number_of_blocks` - can we just always set it to MAX_INT? It may help to document exactly how these intend to be used, though I've left some further motivating comments below.

Reply:
There are three use cases we may be interested in:
1. A full sync 
2. A partial sync
3. Perform an SCID query if we receive an unknown channel_update

Focusing on the full sync, we can drop the `first_blocknum` and `number_of_blocks` arguments and simplify the interface to a single method `perform_routing_sync(&self, their_node_id: &PublicKey)` or similarly named. This could initiate the message flow that covers both channel discovery and request of info. It will depend on how we want to handle state (more on that below).

Reply:
I think you may want to keep `number_of_blocks`, you might have clients with limited storage and thus building their network map iteratively (request channel announcement/updates for a range, discard not-interesting channels, move forward). I agree on the NetworkGraph+chain_hash, though it can be done as a follow-up.

Reply:
Right, I guess my thinking here is in part because we haven't implemented the peer_handler part here, and I don't know how interesting it would be to have a generic API which allows all kinds of requests if we only ever make one or two request types in our own code - my gut says probably not so interesting.